src/rootAstNodesBuilder.ts

```diff
@@ -892,126 +1050,117 @@ function getArguments(tokens, stringArgumentPlaceholders, argumentValidator, val
  * @param stringArgumentPlaceholders The mapping of string literal node argument placeholders to original values.
  * @returns An argument value definition.
  */
-function getArgumentDefinition(token, stringArgumentPlaceholders) {
+function getArgumentDefinition(token: string, stringArgumentPlaceholders: Placeholders): AnyArgument {
     // Check whether the token represents a null value.
     if (token === "null") {
         return {
             value: null,
-            type: "null",
-            toString: function () {
-                return this.value;
-            }
-        };
+            type: "null"
+        } as NullArgument;
     }
 
     // Check whether the token represents a boolean value.
     if (token === "true" || token === "false") {
         return {
             value: token === "true",
-            type: "boolean",
-            toString: function () {
-                return this.value;
-            }
-        };
+            type: "boolean"
+        } as BooleanArgument;
     }
 
     // Check whether the token represents a number value.
-    if (!isNaN(token)) {
+    // TODO: Relies on broken isNaN - see MDN.
+    // if (!Number.isNaN(token)) {
+    if (!isNaN(token as any)) {
```

Comment:
See comment.

I was looking into `isNan` because I wasn't sure it was the right thing to use here. Apparently `Number.isNan` is better.

But when I changed this the tests started failing, and I made an executive decision to ignore the problem and bail. Any ideas if this could cause problems?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN

> The isNaN() function determines whether a value is [NaN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NaN) when converted to a number. Because coercion inside the isNaN() function can be [surprising](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN#description), you may alternatively want to use [Number.isNaN()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN).

See the examples for more